### PR TITLE
ACMS-3477: Add sitestudio_gin module in ACMS.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #[ACMS-3477](https://acquia.atlassian.net/browse/ACMS-3477)

**Proposed changes**
Add sitestudio_gin module in ACMS.

**Alternatives considered**

**Testing steps**
Follow from ticket.

### CODE REVIEW

**Review by deepakmishra2 — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -131,7 +131,8 @@
         "patches": {
             "acquia/acquia-cms-starterkit": {
                 "ACMS-1988 - Update admin theme in starterkit to Gin": "https://github.com/acquia/acquia-cms-starterkit/compare/develop...feature/admin-gin-theme.patch",
-                "ACMS-3456: Update logic to enable theme.": "https://patch-diff.githubusercontent.com/raw/acquia/acquia-cms-starterkit/pull/132.patch"
+                "ACMS-3456: Update logic to enable theme.": "https://patch-diff.githubusercontent.com/raw/acquia/acquia-cms-starterkit/pull/132.patch",
+                "ACMS-4377: Add sitestudio_gin module in ACMS.": "https://patch-diff.githubusercontent.com/raw/acquia/acquia-cms-starterkit/pull/135.patch"
```
Comment (apathak18) [inline composer.json:135]: Not sure whether we should include these patches, CI is passing I guess once vishal review and approve this, we should remove these patches.

File: tests/src/ExistingSite/InstallStateTest.php [hunk 2]
```diff
@@ -241,37 +240,41 @@ private function createMedia(array $values = []) {
...
-
-    // Visit node edit page created above.
-    $this->drupalGet($node_page->toUrl('edit-form'));
-    $assert_session->statusCodeEquals(200);
-
-    // Assert clone tab exists.
-    $assert_session->linkExists('Clone');
+  // Create new content type.
+  $test_node_type = NodeType::create([
+  'type' => 'test_node',
```
Comment (apathak18) [inline tests/src/ExistingSite/InstallStateTest.php:249]: PHPCS issues

File: tests/src/ExistingSite/InstallStateTest.php [hunk 3]
```diff
@@ -241,37 +240,41 @@ private function createMedia(array $values = []) {
    * Check if newly created content type's content can be
    * cloned by user or not.
    */
+  // @todo The action buttons disappeared on the node edit page, isue raised
+  // Issue - https://www.drupal.org/project/gin/issues/3410516
+  /*
```
Comment (apathak18) [inline tests/src/ExistingSite/InstallStateTest.php:245]: @rajeshreeputra If this test isn't working and issue is already there only against gin theme then could it possible to add some logic to work this test method for other themes except gin.

 cc: @vishalkhode1 @chandan-singh7929

File: tests/src/ExistingSiteJavascript/CohesionTestBase.php [hunk 4]
```diff
@@ -32,9 +32,7 @@ protected function setUp(): void {
     parent::setUp();
     $this->moduleInstaller = $this->container->get('module_installer');
     $this->moduleInstaller->install(['sitestudio_claro']);
-    // Set a standard window size so that all javascript tests start with the
-    // same viewport.
-    $this->getDriverInstance()->maximizeWindow();
+    $this->getDriverInstance()->resizeWindow(1920, 800);
```
Comment (apathak18) [inline tests/src/ExistingSiteJavascript/CohesionTestBase.php:35]: Why we are changing this?

File: tests/src/ExistingSiteJavascript/EntityCloneIntegrationTest.php [hunk 5]
```diff
@@ -31,18 +29,21 @@ public function testClone(): void {
...
     $this->drupalGet($edit_form);
     $page = $this->getSession()->getPage();
-    $assert_session = $this->assertSession();
-
+    // $assert_session = $this->assertSession();
     // Add Hero component to the layout canvas.
     $this->getLayoutCanvas()->add('Hero');
     $page->pressButton('Save');
 
+    // @todo The action buttons disappeared on the node edit page, isue raised
```
Comment (apathak18) [inline tests/src/ExistingSiteJavascript/EntityCloneIntegrationTest.php:37]: @rajeshreeputra We can add conditional check to skip this testmethod by using  $this->markTestSkipped
Or apply any better approach

cc: @vishalkhode1

File: tests/src/ExistingSiteJavascript/CohesionTestBase.php (hunk 4 above)
Comment (rajeshreeputra) [reply tests/src/ExistingSiteJavascript/CohesionTestBase.php:35]: It's been reverted to previous status.

File: composer.json (hunk 1 above)
Comment (rajeshreeputra) [reply composer.json:135]: The same procedure is followed: once the pull request against the Acquia CMS Starter Kit is merged, the corresponding patch is removed.

File: tests/src/ExistingSite/InstallStateTest.php (hunk 2 above)
Comment (rajeshreeputra) [reply tests/src/ExistingSite/InstallStateTest.php:249]: There isn't any PHPCS warning/error in CI or on Local.

File: tests/src/ExistingSiteJavascript/EntityCloneIntegrationTest.php (hunk 5 above)
Comment (rajeshreeputra) [reply tests/src/ExistingSiteJavascript/EntityCloneIntegrationTest.php:37]: The issue with gin theme is now fixed in dev release, hence reverted this temp. commit.

File: tests/src/ExistingSite/InstallStateTest.php (hunk 3 above)
Comment (rajeshreeputra) [reply tests/src/ExistingSite/InstallStateTest.php:245]: The issue with gin theme is now fixed in dev release so updated, hence reverted this temp. commit.

**Review by apathak18 — APPROVED** [review]
Patches should be removed before merge

**Review by chandan-singh7929 — APPROVED** [review]

File: modules/acquia_cms_site_studio/composer.json [hunk 6]
```diff
@@ -8,6 +8,7 @@
         "acquia/cohesion-theme": "~7.4.0",
         "drupal/acquia_cms_common": "1.x-dev || 2.x-dev || 3.x-dev",
         "drupal/collapsiblock": "^4.0",
+        "drupal/sitestudio_gin": "^1",
```
Comment (vishalkhode1) [inline modules/acquia_cms_site_studio/composer.json:11]: @rajeshreeputra I don't think we should add this dependency. Let's remove this.
cc: @vipin-mittal-acquia

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: modules/acquia_cms_site_studio/composer.json (hunk 6 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_site_studio/composer.json:11]: @vishalkhode1 Is there a particular reason not to include this functionality by default?

File: composer.json (hunk 1 above)
Comment (deepakmishra2) [inline composer.json:135]: @rajeshreeputra kindly remove this patch too.

Comment (rajeshreeputra) [reply composer.json:135]: this will be removed once [PR](https://github.com/acquia/acquia-cms-starterkit/pull/135) is merged.

Comment (deepakmishra2) [reply composer.json:135]: its merged , we will see if the CI is passing after that or not

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**